target/linux/lantiq/base-files/etc/uci-defaults/02_network_migration

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+#
+# Copyright (C) 2017 OpenWrt.org
+#
+
+. /lib/functions.sh
+. /lib/functions/lantiq.sh
+
+if lantiq_is_vdsl_system; then
+	config_load network
+	config_get iface wan ifname
+	[ "$iface" = "nas0" ] || [ "$iface" = "ptm0" ] || exit
```

Comment:
This will not migrate configs using a vlan tag. You need to check for begins with and replace only the nas0/ptm0 part afterwards.

This should do the job:

```
#!/bin/sh

. /lib/functions.sh
. /lib/functions/lantiq.sh

WAN_IFACE_CHANGED=0

if lantiq_is_vdsl_system; then
	config_load network
	config_get iface wan ifname

	case $iface in
		nas0*)
			iface=${iface/nas0/dsl0}
			WAN_IFACE_CHANGED=1
			;;
		ptm0*)
			iface=${iface/ptm0/dsl0}
			WAN_IFACE_CHANGED=1
			;;
	esac

	uci set network.wan.ifname=$iface
fi

[ "$WAN_IFACE_CHANGED" = "1" ] && uci commit network

exit 0
```
```

Reply:
02_migrate_xdsl_iface *might be* a better name for the script.

Reply:
One more thing. You need to take care of the wan_dev device config as well (the one that sets the mac address). It might work using the old interface name as well, but I prefer to have everything migrated to the same values used by a fresh config.

Reply:
Oh yes, you are right. I've forgot the vlan variants.

Reply:
OK, will also have a look at the wan_dev section